src/ResourceManager/Sql/Commands.Sql/Data Sync/Cmdlet/RemoveAzureSqlSyncAgent.cs

```diff
@@ -0,0 +1,91 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Sql.DataSync.Model;
+
+namespace Microsoft.Azure.Commands.Sql.DataSync.Cmdlet
+{
+    /// <summary>
+    /// Cmdlet to delete a sync agent
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmSqlSyncAgent",
+        SupportsShouldProcess = true)]
+    public class RemoveAzureSqlSyncAgent : AzureSqlSyncAgentCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the sync agent name
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The sync agent name.")]
+        [ValidateNotNullOrEmpty]
+        public string SyncAgentName { get; set; }
+
+        /// <summary>
+        /// Defines whether it is ok to skip the removal confirmation
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Skip confirmation message for performing the action")]
+        public SwitchParameter Force { get; set; }
+
+        /// <summary>
+        ///  Defines whether the cmdlets will output the model object at the end of its execution
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Defines Whether return the removed sync agent")]
+        public SwitchParameter PassThru { get; set; }
+
+        /// <summary>
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
+        /// </summary>
+        /// <returns>True if the model object should be written out, False otherwise</returns>
+        protected override bool WriteResult() { return PassThru; }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlSyncAgentModel> GetEntity()
+        {
+            return new List<AzureSqlSyncAgentModel>() { 
+                ModelAdapter.GetSyncAgent(this.ResourceGroupName, this.ServerName, this.SyncAgentName) 
+            };
+        }
+
+        /// <summary>
+        /// Remove the sync agent
+        /// </summary>
+        /// <param name="entity">The sync agent that this cmdlet operates on</param>
+        /// <returns>The input entity</returns>
+        protected override IEnumerable<AzureSqlSyncAgentModel> PersistChanges(IEnumerable<AzureSqlSyncAgentModel> entity)
+        {
+            ModelAdapter.RemoveSyncAgent(this.ResourceGroupName, this.ServerName, this.SyncAgentName);
+            return entity;
+        }
+
+        /// <summary>
+        /// Entry point for the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            if (!Force.IsPresent && !ShouldProcess(
```

Comment:
This is not the correct way to implement ShouldProcess.  This should be implemented as 
```if (ShouldProcess)
{
  if (Force || ShouldContinue)
  { 
    <execute action>
```

You can get this behavior using ```ConfirmAction(bool, string, string, Action)``` as well